### PR TITLE
added function to create user on auth service from indaba backend

### PR DESCRIPTION
**Pull request by Perry5**
#### What's this PR do?
Adds a function that creates a user on the auth-service whenever that user is created on indaba

#### Related JIRA tickets:
https://jira.amida-tech.com/browse/INBA-385

#### How should this be manually tested?
- Make sure you have the auth-service running
After you have gotten the auth-service running, there are a few ways to test this:
- You can run the `user.integration` test: `node_modules/.bin/mocha test/user.integration.js --bail -t 10000`. Make sure all the tests pass and you can verify that the users were created on the auth-service by looking at the `users` table on the auth-service

- You can run indaba-client, create a project and add users to that project. When you add a user, verify that the user was added to indaba and also the auth service. 

- You can also use postman to hit this end point: `http://localhost:3005/testorg/v0.2/users/`
#### Any background context you want to provide?
Auth service must be running

### CODE REVIEW

File: backend/app/controllers/users.js [hunk 1]
```diff
@@ -1171,3 +1178,33 @@ function* insertOne(req, res, next) {
     }
     return user;
 }
+
+function _createUserOnAuthService(email, password, roleId) {
```
Comment (jsachs) [inline backend/app/controllers/users.js]: @Perry5 are we going to need some sort of callback on this? Either for calling it internally or for potentially using a promise with the created user?

Comment (Perry5) [reply backend/app/controllers/users.js]: Thats a good point. I'll go ahead and make this a callback

**Comment by jmfadeley** [comment]
I'm seeing a very interesting issue where if I get the "User could not be created on the auth service" error, the server shuts down...

**Comment by jmfadeley** [comment]
It works from the front end approach. Basically, Indaba creates a new user. Greyscale accepts, passes the result onto the auth service. User is created there. Can't test any further than that I think because the password is scrambled and we need the reset password system thereafter.

**Comment by jmfadeley** [comment]
This is way better because it no longer crashes the server, but there's problem:

`{
  "!": 0,
  "e": 500,
  "message": "Unable to use auth service: 400 - {\"message\":\"[{\\\"message\\\":\\\"username must be unique\\\",\\\"type\\\":\\\"unique violation\\\",\\\"path\\\":\\\"username\\\",\\\"value\\\":\\\"rs@mail.net\\\"}]\",\"stack\":\"Error: [{\\\"message\\\":\\\"username must be unique\\\",\\\"type\\\":\\\"unique violation\\\",\\\"path\\\":\\\"username\\\",\\\"value\\\":\\\"rs@mail.net\\\"}]\\n    at APIError.ExtendableError (C:\\\\Users\\\\Odin\\\\workspace\\\\amida-auth-microservice\\\\dist\\\\server\\\\helpers\\\\APIError.js:35:15)\\n    at new APIError (C:\\\\Users\\\\Odin\\\\workspace\\\\amida-auth-microservice\\\\dist\\\\server\\\\helpers\\\\APIError.js:63:105)\\n    at C:\\\\Users\\\\Odin\\\\workspace\\\\amida-auth-microservice\\\\dist\\\\config\\\\express.js:123:21\\n    at Layer.handle_error (C:\\\\Users\\\\Odin\\\\workspace\\\\amida-auth-microservice\\\\node_modules\\\\express\\\\lib\\\\router\\\\layer.js:71:5)\\n    at trim_prefix (C:\\\\Users\\\\Odin\\\\workspace\\\\amida-auth-microservice\\\\node_modules\\\\express\\\\lib\\\\router\\\\index.js:315:13)\\n    at C:\\\\Users\\\\Odin\\\\workspace\\\\amida-auth-microservice\\\\node_modules\\\\express\\\\lib\\\\router\\\\index.js:284:7\\n    at Function.process_params (C:\\\\Users\\\\Odin\\\\workspace\\\\amida-auth-microservice\\\\node_modules\\\\express\\\\lib\\\\router\\\\index.js:335:12)\\n    at Immediate.next (C:\\\\Users\\\\Odin\\\\workspace\\\\amida-auth-microservice\\\\node_modules\\\\express\\\\lib\\\\router\\\\index.js:275:10)\\n    at Immediate.<anonymous> (C:\\\\Users\\\\Odin\\\\workspace\\\\amida-auth-microservice\\\\node_modules\\\\express\\\\lib\\\\router\\\\index.js:635:15)\\n    at runCallback (timers.js:639:20)\\n    at tryOnImmediate (timers.js:610:5)\\n    at processImmediate [as _immediateCallback] (timers.js:582:5)\"}"
}`

We get this callback should the user already be in the auth DB. The user being in the DB is fine, but because of this error:
1) The user isn't added to the project at all.
2) We don't report the addition to the user via the "registered" field from the return. 

We need to fix that. I can either accept and merge this, and I'll create a new PR ticket, or you can fix it here.

**Comment by jmfadeley** [comment]
@Perry5 I pushed a possible fix for the duplicate error coming back from auth. Those return a 400, so we look for that and basically let the process run as normal. Anything else, 500 localized internal error like you were doing. Try it, change if you want.

**Comment by Perry5** [comment]
@jmfadeley this works fine. I noticed that you can add a user with the same email to a project twice. We might need to add another check on the front end for that. But I'm going ahead to merge this in. We need to create a ticket to write tests mocking the auth service and creating users.

**Pull request merged**